client/src/components/app/App.js

```diff
@@ -1,34 +1,61 @@
 import "./App.css";
 import { Switch, Route, NavLink } from "react-router-dom";
+import axios from "axios";
+import { useState, useEffect } from "react";
 import AppHeader from "./AppHeader.js";
-import HomePage from "../../pages/HomePage.js";
-// import AboutPage from "../../pages/AboutPage.js";
-import FavsPage from "../../pages/FavsPage.js";
-import BooksPage from "../../pages/BooksPage.js";
-import BookPage from "../../pages/BookPage.js";
+
+import MyCommunitiesPage from "../../pages/MyCommunitiesPage.js";
+
+import CommunitiesContext from "../../contexts/CommunitiesContext";
+
+import Modal from "../modal/Modal.js";
+import NewCommunityModalPage from "../../pages/NewCommunityModalPage.js";
 
 function App() {
+  const [resources, setResources] = useState([]);
+
+  const primaryAddActionFormId = "newCommunity";
+
+  useEffect(() => {
+    console.log("yohooo effect");
+    axios
+      .get("/api/communities")
+      .then((res) => {
+        setResources(res.data || []);
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  }, []);
+
   return (
     <div className="App">
-      <AppHeader />
-      <main>
-        <Switch>
-          <Route exact path="/" component={HomePage} />
-          <Route path="/books/:id" component={BookPage} />
-          <Route path="/books" component={BooksPage} />
-          {/* <Route path="/about" component={AboutPage} /> */}
-          <Route path="/favs" component={FavsPage} />
-          <Route path="*">
-            <h2>
-              404 - Sorry, my friend of horror, but this page could not be
-              found!
-            </h2>
-          </Route>
-        </Switch>
-      </main>
-      <footer>
-        <div>Made with love for books by Chris</div>
-      </footer>
+      <CommunitiesContext.Provider value={{ resources, setResources }}>
```

Comment:
I would recommend not passing the pure `setResources` function here. I would create a function here in this component, which is just doing what is supposed to do. For example, updating by ID, or deleting by ID. Right now, you can do everything with `setResources`. Do you want that? ;-)

Reply:
In the encapsulated components, I just pick what I need from the surrounding context - either `resource`or `setResource`  which works like a charm - but: If it's 'react-convention' to let a parent function do all the heavy CRUD-lifting in dedicated functions and pass those down to the relevant child-components as props than I'll go with it - but (the second): How do I handle the foreseeable situation when a multitude of different models (say user, user-groups, posts and whatnot next to communities) need to be rendered in CRUD-components starting from App.js - how do I split this in different files on App.js-level?